repose-aggregator/components/filters/uri-user-filter/src/main/scala/org/openrepose/filters/uriuser/UriUserFilter.scala

```diff
@@ -0,0 +1,105 @@
+import org.openrepose.filters.uriuser.config.UriUserConfig
+
+import scala.collection.JavaConversions._
+
+@Named
+class UriUserFilter @Inject()(configurationService: ConfigurationService)
+  extends Filter with UpdateListener[UriUserConfig] with LazyLogging {
+
+  import UriUserFilter._
+
+  private var configurationFileName: String = DefaultConfigFileName
+  private var initialized = false
+  private var config: Config = _
+
+  override def init(filterConfig: FilterConfig): Unit = {
+    logger.trace("URI User filter initializing...")
+    configurationFileName = new FilterConfigHelper(filterConfig).getFilterConfig(DefaultConfigFileName)
+
+    logger.info(s"Initializing URI User Filter using config $configurationFileName")
+    val xsdUrl: URL = getClass.getResource(SchemaFileName)
+    configurationService.subscribeTo(filterConfig.getFilterName, configurationFileName, xsdUrl, this, classOf[UriUserConfig])
+
+    logger.trace("URI User filter initialized.")
+  }
+
+  override def doFilter(servletRequest: ServletRequest, servletResponse: ServletResponse, filterChain: FilterChain): Unit = {
+    if (!isInitialized) {
+      logger.error("URI User filter has not yet initialized...")
+      servletResponse.asInstanceOf[HttpServletResponse].sendError(HttpServletResponse.SC_SERVICE_UNAVAILABLE)
+    } else {
+      val wrappedRequest = new HttpServletRequestWrapper(servletRequest.asInstanceOf[HttpServletRequest])
+
+      config.uriRegex.toStream
+        .map(_.matcher(wrappedRequest.getRequestURI))
+        .find(m => m.find() && m.groupCount > 0)
+        .map(_.group(1)) foreach { user =>
+          wrappedRequest.addHeader(PowerApiHeader.USER.toString, user, config.quality)
+          wrappedRequest.addHeader(PowerApiHeader.GROUPS.toString, config.group, config.quality)
+      }
+
+      filterChain.doFilter(wrappedRequest, servletResponse)
+    }
+  }
+
+  override def destroy(): Unit = {
+    logger.trace("URI User filter destroying...")
+    configurationService.unsubscribeFrom(configurationFileName, this.asInstanceOf[UpdateListener[_]])
+    logger.trace("URI User filter destroyed.")
+  }
+
+  override def configurationUpdated(uriUserConfig: UriUserConfig): Unit = {
+    config = Config(
+      uriUserConfig.getIdentificationMappings.getMapping.map(_.getIdentificationRegex.r.pattern),
+      Option(uriUserConfig.getGroup).filterNot(_.trim.isEmpty).getOrElse(DefaultGroup),
+      Option(uriUserConfig.getQuality).map(_.toDouble).getOrElse(DefaultQuality))
+    initialized = true
+  }
+
+  override def isInitialized: Boolean = initialized
+}
+
+object UriUserFilter {
+  private final val DefaultConfigFileName = "uri-user.cfg.xml"
+  private final val SchemaFileName = "/META-INF/schema/config/uri-user-configuration.xsd"
+
+  // while the XSD specifies these defaults, the generated getters don't provide these values for some reason
+  private final val DefaultQuality = 0.5d
+  private final val DefaultGroup = "User_Standard"
```

Comment:
I keep forgetting that the Scala convention is to capitalize the first letter of each word for constants. If you happen to review my code in the future and I've capitalized every letter, please correct me! =)